go-controller/pkg/ovn/pods.go

```diff
@@ -159,44 +159,43 @@ func (oc *Controller) deleteLogicalPort(pod *kapi.Pod) {
 		annotation, err := util.UnmarshalPodAnnotation(pod.Annotations)
 		if err == nil {
 			podIfAddrs := annotation.IPs
-			_ = oc.lsManager.ReleaseIPs(pod.Spec.NodeName, podIfAddrs)
+			err = oc.lsManager.ReleaseIPs(pod.Spec.NodeName, podIfAddrs)
 		}
-		return
+		return err
 	}
 
 	// FIXME: if any of these steps fails we need to stop and try again later...
 	var allOps, ops []ovsdb.Operation
 	if ops, err = oc.deletePodFromNamespace(pod.Namespace, portInfo); err != nil {
-		klog.Errorf(err.Error())
-	} else {
-		allOps = append(allOps, ops...)
+		return err
 	}
+	allOps = append(allOps, ops...)
 
 	ops, err = oc.ovnNBLSPDel(logicalPort, pod.Spec.NodeName, portInfo.uuid)
 	if err != nil {
-		klog.Errorf(err.Error())
-	} else {
-		allOps = append(allOps, ops...)
+		return err
 	}
+	allOps = append(allOps, ops...)
 
 	_, err = libovsdbops.TransactAndCheck(oc.nbClient, allOps)
 	if err != nil {
-		klog.Errorf("Cannot delete logical switch port %s, %v", logicalPort, err)
+		return fmt.Errorf("cannot delete logical switch port %s, %v", logicalPort, err)
 	}
 
 	if err := oc.lsManager.ReleaseIPs(portInfo.logicalSwitch, portInfo.ips); err != nil {
-		klog.Errorf(err.Error())
+		return err
 	}
 
 	if config.Gateway.DisableSNATMultipleGWs {
 		if err := deletePerPodGRSNAT(oc.nbClient, pod.Spec.NodeName, []*net.IPNet{}, portInfo.ips); err != nil {
-			klog.Errorf(err.Error())
+			return err
 		}
 	}
 	podNsName := ktypes.NamespacedName{Namespace: pod.Namespace, Name: pod.Name}
 	oc.deleteGWRoutesForPod(podNsName, portInfo.ips)
 
 	oc.logicalPortCache.remove(logicalPort)
```

Comment:
following the question from @flavio-fernandes, would be safe to continue in case the cache for some reason is not able to be deleted?  If I am looking to the right place, there is a return error from delete:
https://github.com/ovn-org/ovn-kubernetes/blob/f5e9d65ac238edd4b3f835e9eed9b5193338f647/go-controller/pkg/libovsdbops/model_client.go#L345

Reply:
So its not the libovsdbops model we are using here. Its:

```
func (c *portCache) remove(logicalPort string) {
	c.Lock()
	defer c.Unlock()
	info, ok := c.cache[logicalPort]
	if !ok || !info.expires.IsZero() {
		klog.V(5).Infof("port-cache(%s): port not found in cache or already marked for removal", logicalPort)
		return
	}
	info.expires = time.Now().Add(time.Minute)
	klog.V(5).Infof("port-cache(%s): scheduling port for removal at %v", logicalPort, info.expires)

	// Removal must be deferred since, for example, NetworkPolicy pod handlers
	// may run after the main pod handler and look for items in the cache
	// on delete events.
	go func() {
		select {
		case <-time.After(time.Minute):
			c.Lock()
			defer c.Unlock()
			// remove the port info if its expiration time has elapsed.
			// This makes sure that we don't prematurely remove a port
			// that was deleted and re-added before the timer expires.
			if info, ok := c.cache[logicalPort]; ok && !info.expires.IsZero() {
				if time.Now().After(info.expires) {
					klog.V(5).Infof("port-cache(%s): removing port", logicalPort)
					delete(c.cache, logicalPort)
				}
			}
		case <-c.stopChan:
			break
		}
	}()
}
```
so in the end its a delete from a golang map.